ctapipe/tools/tests/test_apply_modesl.py

```diff
@@ -0,0 +1,138 @@
+from ctapipe.containers import (
+    EventIndexContainer,
+    ParticleClassificationContainer,
+    ReconstructedEnergyContainer,
+)
+from ctapipe.core import run_tool
+from ctapipe.io import TableLoader, read_table
+
+
+def test_apply_energy_regressor(
+    energy_regressor_path,
+    dl2_shower_geometry_file_lapalma,
+    tmp_path,
+):
+    from ctapipe.tools.apply_models import ApplyModels
+
+    input_path = dl2_shower_geometry_file_lapalma
+    output_path = tmp_path / "energy.dl2.h5"
+
+    ret = run_tool(
+        ApplyModels(),
+        argv=[
+            f"--input={input_path}",
+            f"--output={output_path}",
+            f"--regressor={energy_regressor_path}",
+            "--Apply.StereoMeanCombiner.weights=konrad",
+        ],
+        raises=True,
+    )
+    assert ret == 0
+
+    table = read_table(output_path, "/dl2/event/subarray/energy/ExtraTreesRegressor")
+    for col in "obs_id", "event_id":
+        assert table[col].description == EventIndexContainer.fields[col].description
+
+    for name, field in ReconstructedEnergyContainer.fields.items():
+        colname = f"ExtraTreesRegressor_{name}"
+        assert colname in table.colnames
+        assert table[colname].description == field.description
+
+    loader = TableLoader(output_path, load_dl2=True)
+    events = loader.read_subarray_events()
```

Comment:
Should we test that these are not (all) nan aswell here (and in the other tests)?